weatherAplication/Modules/ShowWeather/ViewModel/ShowWeatherViewModel.swift

```diff
@@ -0,0 +1,17 @@
+import Foundation
+import UIKit
+
+class ShowWeatherViewModel {
+
+    var delegate: ShowWeatherDelegate?
+
+    func chosenImage(weatherCode: WeatherCodes) -> UIImage? {
+
+        delegate?.chosenImage(weatherCode: weatherCode)
+
+    }
+}
```

Comment:
Tutaj też jest dokładnie ta sama sytuacja jak w SearchViewModelu, to znaczy w tym miejscu powinno być pobieranie informacji z API i ich dekodowanie, i dopiero poprzez delegata przekazywanie tych przerobionych danych do viewControllera.